test/Microsoft.AspNet.Mvc.ModelBinding.Test/Validation/ModelStateDictionaryTest.cs

```diff
@@ -733,6 +734,53 @@ public void ModelStateDictionary_NoErrorMessage_ForNonFormatException()
             Assert.Empty(error.ErrorMessage);
         }
 
+        [Fact]
+        public void ModelStateDictionary_ClearEntriesPrefixedWithKey_NonEmptyKey()
```

Comment:
Also a case there the string is a prefix, but it's not considered a model prefix.

Ex:
`Bro` clears `Bro` and `Bro.Chill` but not `Broseidon`